core/kazoo_couch/src/kz_couch_attachments.erl

```diff
@@ -114,9 +114,25 @@ relay_stream_attachment(Caller, Ref, Msg) ->
 -spec do_put_attachment(couchbeam_db(), ne_binary(), ne_binary(), ne_binary(), kz_proplist()) ->
                                {'ok', kz_json:object()} |
                                couchbeam_error().
-do_put_attachment(#db{}=Db, DocId, AName, Contents, Options) ->
+do_put_attachment(#db{}=Db, DocId, AName, Contents0, Options0) ->
+    Contents = {fun stream_att/1, Contents0},
+    Options = [{send_timeout, 20000}
+              ,{headers, [{<<"Transfer-Encoding">>, <<"chunked">>}
+                         ,{<<"Expect">>, <<"100-Continue">>}
+                         ]}
+              ,{content_length, erlang:byte_size(Contents0)}
+               | Options0
+              ],
     ?RETRY_504(couchbeam:put_attachment(Db, DocId, AName, Contents, Options)).
 
+-spec stream_att(binary()) -> {'ok', binary(), binary()} | 'eof'.
+stream_att(<<Data:16384/binary, Rest/binary>>) ->
```

Comment:
We should probably make this length configurable, shouldn't we?
Also I restarted the mk-bs build